File: Python/271_Encode_and_Decode_Strings.py
from typing import List

class Solution:

    def encode(self, strs: List[str]) -> str:
        res=""
        sizes=[]
        for i in strs:
            res+=str(len(i))+"#"+i
        return res

    def decode(self, s: str) -> List[str]:
        res=[]
        word=""
        i=0
        while i <len(s):
            j=i
            while s[j]!='#':
                j+=1
            size=int(s[i:j])
            i=j+1
            word=s[i:i+size]
            res.append(word)
            i+=size
        return res


test_case = ["neet", "code", "love", "you"]

solution = Solution()
encoded = solution.encode(test_case)
decoded = solution.decode(encoded)

print("Encoded:", encoded)
print("Decoded:", decoded)

File: Java/271_Encode_and_Decode_Strings.java
class Solution {

    public String encode(List<String> strs) {
        StringBuilder res=new StringBuilder();
        for(String s: strs){
            res.append(s.length()).append("#").append(s);
        }
        return res.toString();
    }

    public List<String> decode(String str) {
        ArrayList<String> res=new ArrayList<String>();
        int index=0;
        while(index<str.length()){
            String s="";
            while(index<str.length() && Character.isDigit(str.charAt(index))){
                s+=str.charAt(index);
                index++;
            }
            if (s.isEmpty()) {
                throw new IllegalArgumentException("Invalid encoded string format.");
            }
            int size=Integer.parseInt(s);
            index++;//remove #
            res.add(str.substring(index,index+size));
            index+=size;
        }
        return res;

    }
}
